ucp/useraccounts_module.php

```diff
@@ -203,13 +159,14 @@ function main($id, $mode)
 			case 'manage_keys':
 			{
 				// Set up vars for mode.
-				$start = $this->request->variable('start', 0); // Used for pagination.
-				$markedkeys = $this->request->variable('markedkeys', array(0)); // items ids.
-				$revokemark = $this->request->is_set_post('markedkeys') && $action == 'rev_marked' ? true : false;
-				$revokeall = $action == 'rev_all' ? true : false;
+				$start = (int)$this->request->variable('start', 0); // Used for pagination.
+				$markedkeys = (array)$this->request->variable('markedkeys', array(0)); // items ids.
+				$revokemark = (bool)$this->request->is_set_post('rev_marked');
```

Comment:
tbh, i'm not sure this is better than what you had before.  There can only be one action at a time, so having a single variable to define which one makes sense.  Having a separate variable for each action can open up edge cases that a malicious actor may be able to abuse.

Reply:
The $action pattern seems to appear in two usecases:
* In a switch statement where the actions are the cases (not suitable for our module with many different modes).
* When parameters are passed through the URL (we use this for ordering).
* When the confirm box is used and there is more than one possible submit "action" to confirm.

Unlike the other phpBB sections, the pattern is not as much used in the UCP context. Also, it is not mentioned in the dedicated section of [Using phpBB3's Basic Functions](https://wiki.phpbb.com/Using_phpBB3%27s_Basic_Functions#1.4.1._User_input). 

To account for your comment, I simplified all submit actions that appear in a context where only one submit actions is possible to a $submit var.

I think the current method is cleaner then the $action method which would also require to define the $action for each sumbit. Additionally, the check if the request was send via POST is done for the submit that was requested instead of checking if a random request variable in the context of the action was send via POST.

Reply:
However, I will check if the `$action` method  can be applied to match URL parameters too.